crates/bestool/src/actions/crypto/keygen.rs

```diff
@@ -0,0 +1,34 @@
+use std::path::PathBuf;
+
+use age::{secrecy::ExposeSecret, x25519};
+use clap::Parser;
+use miette::{IntoDiagnostic as _, Result};
+
+use crate::actions::{crypto::CryptoArgs, Context};
+
+#[derive(Debug, Clone, Parser)]
+pub struct KeygenArgs {
+	/// The destination directory the output will be written to.
+	#[cfg_attr(docsrs, doc("\n\n**Flag**: `--output PATH`"))]
+	#[arg(long, default_value = r"./")]
+	pub output: PathBuf,
+}
+
+pub async fn run(ctx: Context<CryptoArgs, KeygenArgs>) -> Result<()> {
+	let output = ctx.args_sub.output;
+	let secret = x25519::Identity::generate();
+	let public = secret.to_public();
+
+	tokio::fs::write(
+		output.join("private_key.txt"),
+		secret.to_string().expose_secret().as_bytes(),
+	)
+	.await
+	.into_diagnostic()?;
+
+	tokio::fs::write(output.join("public_key.txt"), public.to_string().as_bytes())
```

Comment:
Hmm I would expect to write the age single-keyfile format:

```
# created: 2023-10-14T17:25:45+01:00
# public key: age1kk84zfvr4whsfdmrg2hk8pf7l3v705a8x9k5khuh0lmtrnuaz4gqnwty3q
AGE-SECRET-KEY-1USUPGW9GDQAQEHM8X3JKQSJMN7YY5T26PXWFHWMRR4Y9TDAHMA5SJDRUUJ
```

defaulting to `./identity.txt`, but with the argument being the full path including filename.